tests/components/octoprint/test_button.py

```diff
@@ -192,3 +192,78 @@ async def test_stop_job(hass: HomeAssistant) -> None:
         )
 
         assert len(stop_command.mock_calls) == 0
+
+
+async def test_shutdown_system(hass: HomeAssistant) -> None:
+    """Test the shutdown system button."""
+    await init_integration(hass, BUTTON_DOMAIN)
+
+    coordinator: OctoprintDataUpdateCoordinator = hass.data[DOMAIN]["uuid"][
+        "coordinator"
+    ]
+
+    # Test shutting down the system
+    with patch("pyoctoprintapi.OctoprintClient.shutdown") as shutdown_command:
+        coordinator.data["printer"] = OctoprintPrinterInfo(
+            {"state": {"flags": {}}, "temperature": []}
+        )
```

Comment:
Is this needed? Looking at the init_integration function, it by defaults set to default printer, which has these attributes with a bit more. You also should not do anything to the coordinator inside of tests. You instead should patch when you need the coordinator to change values